research/conditional/utils/check_args.py

```diff
@@ -38,3 +38,9 @@ def check_args(args):
 
     if args.repeater_mode:
         assert args.relative_init_scale == None
+
+    assert args.lr_trapezoidal_decay_percent + args.lr_warmup_percent <= 1
+    assert (
+        args.lr_trapezoidal_decay_percent + (args.lr_warmup_steps / args.final_lr_step)
+        <= 1
+    )
```

Comment:
there is a function check_args in research/conditional/utils/check_args.py maybe it's a better place for asserts like this one? also final_lr_step can be -1 or None

Reply:
changed